x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/AggregatorFunctionTestCase.java

```diff
@@ -198,7 +197,6 @@ public final void testNullIntermediateFinal() {
      * return other sorts of results.
      */
     protected void assertOutputFromEmpty(Block b) {
-        assertThat(b.elementType(), equalTo(ElementType.NULL));
```

Comment:
"Empty" means a a block with a null. So, there are 2 ways to create a block with a null: A (constant) Null block, or a <type> block. Both support having that null there.

For this aggregation (And I believe for some others too), building a null block means short-circuiting based on some check. That check may have a cost. In the case of this aggregation, it was either keeping a flag, or "deducing" that an empty array (In BucketedSort) meant no buckets, ergo no values. So some extra logics here.

Context aside: I removed it because it looked over-fitting, and as it's called on aggregation completion. I checked the difference between a constant null block and a long block, and yeah, the long blocks allocates an extra array. So I wonder if that's the reason for this check.

_I wanted to comment this, but maybe I forgot about it. Thanks for pointing it out!_

Reply:
Got it. I'm beginning to remember why I had these. I should have left a comment a year ago. If wishes were fishes then... how does the phrase go?

Anyway, constant null blocks have some neat tricks, not so much in the compute engine, but we reuse them. I had been working on some optimizations when you have an index with a zillion fields but they are empty - and get loaded as null blocks. We reuse the same instance for these blocks a bunch and the deserialization is a bit faster for it. And a little less memory. That's important when there's like 10000 of the things. Which happens. A lot more than I'd like.

So aggs producing the full null block are less important than fields loading them. But it does feel useful to make the `null`-only block.

Reply:
Ok! Adding the logic then, and re-adding that check